backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/work/impl/SearchWork.java

```diff
@@ -127,6 +126,14 @@ private Builder(JsonObject payload, ElasticsearchSearchResultExtractor<R> result
 			return this;
 		}
 
+		@Override
+		public SearchWorkBuilder<R> disableTrackTotalHits() {
+			if ( trackTotalHits != null && trackTotalHits ) {
```

Comment:
Why this check? Someone is asking to disable track total hits, we should just do it. No need for complex logic here.

Reply:
In the current (master) behavior `trackTotalHits` is null in case the version of Elasticsearch is strictly less than 7.
If we tried to set `track_total_hits` in ES 6, we got an error, since the parameter is not supposed to be set.
I just kept the preexisting convention, setting the parameter to `false` only if the parameter was previously set. 
I agree with you that it is not very expressive.

Reply:
I'm adding a comment, since other changes could have a larger impact...